core-container/entry.sh

```diff
@@ -43,56 +33,78 @@ function set_provider_adapters() {
     rm kcpsecret
 
     echo "Delete the kcp provider object if it already exists, and then create it."
-    if ! kubectl delete spaceproviderdesc pkcp ; then
+    if ! kubectl delete spaceproviderdesc $PROVIDER_NAME ; then
         echo "Nothing to delete."
     fi
     kubectl create -f - <<EOF
 apiVersion: space.kubestellar.io/v1alpha1
 kind: SpaceProviderDesc
 metadata:
-  name: pkcp
+  name: $PROVIDER_NAME
 spec:
   ProviderType: "kcp"
   SpacePrefixForDiscovery: "ks-"
   secretRef:
     namespace: ${NAMESPACE}
     name: kcpsec
 EOF
-
-    if [ "$SPACE_PROVIDER" == "kubeflex" ]; then
-        echo "Waiting for the kubeflex provider to be ready... this may take a while."
-        (
-            until [ "$(kubectl get pods -A | grep kubeflex-controller-manager | grep Running)" != "" ]; do
-                sleep 10
-            done
-        )
+}
  
-        echo "Delete the kubeflex provider object if it already exists, and then create it."
-        if ! kubectl delete spaceproviderdesc pkflex ; then
-            echo "Nothing to delete."
-        fi
-        kubectl create -f - <<EOF
+function create_kubeflex_provider() {
+    echo "Waiting for the kubeflex provider to be ready... this may take a while."
+    (
+        until [ "$(kubectl get pods -A | grep kubeflex-controller-manager | grep Running)" != "" ]; do
+            sleep 10
+        done
+    )
+
+    echo "Delete the kubeflex provider object if it already exists, and then create it."
+    if ! kubectl delete spaceproviderdesc $PROVIDER_NAME ; then
+        echo "Nothing to delete."
+    fi
+    kubectl create -f - <<EOF
 apiVersion: space.kubestellar.io/v1alpha1
 kind: SpaceProviderDesc
 metadata:
-  name: pkflex
+  name: $PROVIDER_NAME
 spec:
   ProviderType: "kubeflex"
   SpacePrefixForDiscovery: "ks-"
   secretRef:
     namespace: ${NAMESPACE}
     name: corecluster
 EOF
-    fi
+    echo "Waiting for default spaceprovider to reach the Ready phase."
+    until [ "$(kubectl --kubeconfig ${SPACE_MANAGER_KUBECONFIG} get spaceproviderdesc $PROVIDER_NAME -o yaml | grep Ready )" != "" ]; do
+        sleep 1
+    done 
+}
+
 
+function set_provider_adapters() {
+    oldKubeconfig=$KUBECONFIG
```

Comment:
There is a simpler way to avoid side-effecting the caller's definition of KUBECONFIG. And, as a bonus, it is also resilient against any and all unexpected ways to exit the function.  The following bash session illustrates the point.


```
bash-3.2$ function test1() { KUBECONFIG=fubar; }
bash-3.2$ function test2() ( KUBECONFIG=baz )
bash-3.2$ KUBECONFIG=something
bash-3.2$ test1
bash-3.2$ echo $KUBECONFIG
fubar
bash-3.2$ test2
bash-3.2$ echo $KUBECONFIG
fubar
```

Reply:
Great idea.  I'll implement.

Reply:
I do not see it implemented here.

Reply:
The implementation has changed such that we no longer keep an old copy of the kubeconfig.